kinto/plugins/accounts/views.py

```diff
@@ -138,3 +223,192 @@ def on_account_changed(event):
     cache_key = utils.hmac_digest(hmac_secret, ACCOUNT_CACHE_KEY.format(username))
     # Delete cache
     cache.delete(cache_key)
+
+
+# Account validation (enable in the settings).
+validation = Service(
+    name="account-validation",
+    path="/accounts/{user_id}/validate/{activation_key}",
+    description="Validate an account",
+)
+
+
+def cache_validation_key(activation_key, username, registry):
+    """Store a validation_key in the cache."""
+    settings = registry.settings
+    hmac_secret = settings["userid_hmac_secret"]
+    cache_key = utils.hmac_digest(hmac_secret, ACCOUNT_VALIDATION_CACHE_KEY.format(username))
+    # Store an activation key for 7 days by default.
+    cache_ttl = int(
+        settings.get("account_validation.validation_key_cache_ttl_seconds", 7 * 24 * 60 * 60)
+    )
+
+    cache = registry.cache
+    cache.set(cache_key, activation_key, ttl=cache_ttl)
+
+
+def check_validation_key(activation_key, username, registry):
+    """Given a username, compare the activation-key provided with the one from the cache."""
+    hmac_secret = registry.settings["userid_hmac_secret"]
+    cache_key = utils.hmac_digest(hmac_secret, ACCOUNT_VALIDATION_CACHE_KEY.format(username))
+
+    cache = registry.cache
+    cache_result = cache.get(cache_key)
+
+    if cache_result == activation_key:
+        cache.delete(cache_key)  # We're done with the activation key.
+        return True
+    return False
+
+
+@validation.post()
+def post_validation(request):
+    user_id = request.matchdict["user_id"]
+    activation_key = request.matchdict["activation_key"]
+
+    parent_id = user_id
+    try:
+        user = request.registry.storage.get(
+            parent_id=parent_id, resource_name="account", object_id=user_id
+        )
+    except storage_exceptions.ObjectNotFoundError:
+        # Don't give information on the existence of a user id: return a generic error message.
+        error_details = {"message": "Account ID and activation key do not match"}
+        raise http_error(httpexceptions.HTTPForbidden(), **error_details)
+
+    if is_validated(user):
+        error_details = {"message": f"Account {user_id} has already been validated"}
+        raise http_error(httpexceptions.HTTPForbidden(), **error_details)
```

Comment:
Maybe we shouldn't care about this. It allows me to know that user X is validated or not